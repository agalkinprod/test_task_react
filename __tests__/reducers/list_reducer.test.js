import listReducer from '../../src/reducers/list_reducer';
import {
    SET_LIST_ITEMS,
    EDIT_LIST_ITEM,
    DELETE_LIST_ITEM
} from '../../src/actions/types';

describe('list_reducer', () => {

  const list_items_mock = [
    { id: 0, text: 'first', is_selected: false },
    { id: 1, text: 'second', is_selected: true },
    { id: 2, text: 'third', is_selected: false }
  ];

  it('should return proper default state', () => {
    const action = { type: 'not_existing_action' };
    expect(listReducer(undefined, action)).toMatchSnapshot();
  });

  it('should set list items', () => {
    const action = { type: SET_LIST_ITEMS, payload: list_items_mock };
    expect(listReducer(undefined, action)).toMatchSnapshot();
  });

  it('should edit list items', () => {
    const updated_list_item = { id: 1, text: 'second', is_selected: false };
    const action = { type: EDIT_LIST_ITEM, payload: updated_list_item };
    expect(listReducer({ list_items: list_items_mock }, action)).toMatchSnapshot();
  });

  it('should delete list item', () => {
    const id_to_delete = 1;
    const action = { type: DELETE_LIST_ITEM, payload: id_to_delete };
    expect(listReducer({ list_items: list_items_mock }, action)).toMatchSnapshot();
  });

});