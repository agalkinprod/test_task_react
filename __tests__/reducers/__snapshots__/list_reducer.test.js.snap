// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list_reducer should delete list item 1`] = `
Object {
  "list_items": Array [
    Object {
      "id": 0,
      "is_selected": false,
      "text": "first",
    },
    Object {
      "id": 2,
      "is_selected": false,
      "text": "third",
    },
  ],
}
`;

exports[`list_reducer should edit list items 1`] = `
Object {
  "list_items": Array [
    Object {
      "id": 0,
      "is_selected": false,
      "text": "first",
    },
    Object {
      "id": 1,
      "is_selected": false,
      "text": "second",
    },
    Object {
      "id": 2,
      "is_selected": false,
      "text": "third",
    },
  ],
}
`;

exports[`list_reducer should return proper default state 1`] = `
Object {
  "list_items": Array [],
}
`;

exports[`list_reducer should set list items 1`] = `
Object {
  "list_items": Array [
    Object {
      "id": 0,
      "is_selected": false,
      "text": "first",
    },
    Object {
      "id": 1,
      "is_selected": true,
      "text": "second",
    },
    Object {
      "id": 2,
      "is_selected": false,
      "text": "third",
    },
  ],
}
`;
